sympy/solvers/solveset.py

```diff
@@ -978,6 +985,331 @@ def _solveset(f, symbol, domain, _check=False):
+    `solveset` and should be used internally. `\_transolve`
+    currently supports the following class of equations:
+
+        - Exponential equations
+
+    Parameters
+    ==========
+
+    f : is any transcendental equation that needs to be solved.
+        This needs to be an expression, which is assumed
+        to be equal to 0.
+
+    symbol : is variable for which the equation is solved.
+        This needs to be of class `Symbol`.
+
+    domain : is a set over which the equation is solved.
+        This needs to be of class `Set`.
+
+    How to use `\_transolve`
+    ========================
+
+    `\_transolve` should not be used as an independent function, because
+    it assumes that the equation (`f`) and the `symbol` comes from
+    `solveset` and might have undergone a few modification(s).
+    To use `\_transolve` as an independent function the equation (`f`)
+    and the `symbol` should be passed as they would have been by
+    `solveset`.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.solveset import _transolve as transolve
+    >>> from sympy.solvers.solvers import _tsolve as tsolve
+    >>> from sympy import symbols, S, pprint
+    >>> # symbol modified as would have been by solveset
+    >>> x = symbols('x', real=True)
+    >>> transolve(5**(x - 3) - 3**(2*x + 1), x, S.Reals)
+    {-log(375)/(-log(5) + 2*log(3))}
+
+    How `\_transolve` works
+    =======================
+
+    `\_transolve` uses two types of helper functions to solve equations
+    of a particular class:
+
+    Identifying helpers: These helpers are used to determine whether a
+    given equation belongs to a certain class of equation or not.
+    These functions return either True or False.
+
+    Solving helpers: These are the helpers that solve the equation.
+    They are invoked once the equation is identified by the `identifying
+    helpers`. These functions return either the complete solution or
+    a modified version of the equation that `solveset` might better
+    handle.
+
+    * Philosophy behind the module
+
+    `\_transolve` comes into action when `solveset` is unable to solve
+    the equation as a last resort to get the solutions. The idea is to
+    get the equation in a case (Add, Pow etc.) that satisfies its
+    general form, for example exponential equations take the form as
+    `a*f(x) + b*g(x)`, where `f(x)` and `g(x)` are power terms,
+    therefore they are included in `Add` case.
+
+    Once done different `identifying helper` conditions are included
+    to check which class the equation belongs. `Solving helper` is
+    invoked once the class of the equation is identified. If the
+    equation is solved the result is returned otherwise a
+    `ConditionSet` is returned.
+
+    How `\_transolve` is better than `\_tsolve`
+    ===========================================
+
+    1) Better output
+
+    `\_transolve` provides a better output to some equations than
```

Comment:
`\_transolve` provides expressions in a more simplified form:

(and delete the rest)